db/models/articles.models.js

```diff
@@ -32,12 +31,24 @@ exports.fetchArticles = () => {
   LEFT JOIN (
     SELECT article_id, COUNT(comment_id) AS comment_count
     FROM comments GROUP BY article_id)
-    c ON a.article_id = c.article_id
-    ORDER BY created_at DESC;`
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+    c ON a.article_id = c.article_id `;
+
+  if (filterBy.includes(filter)) {
+    queryVals.push(query);
+    queryString += `WHERE ${filter}=$1 `;
+  }
+  queryString += `ORDER BY created_at DESC;`;
+
+  return db.query(queryString, queryVals).then(({ rows }) => {
+    if ( !rows.length)
```

Comment:
Once we incorporate the new test I mentioned above you will likely need to remove this error handling from here as an empty response we will find could potentially be valid. You will likely need to check the topic before this query somehow. Similar to how we checked for an article existing but having no comments.